docs/README.new-provider.md

```diff
@@ -0,0 +1,19 @@
+# Implementing a Provider for Secrets Store CSI Driver
+
+This document highlights the implementation steps for adding a secrets-store-csi-driver provider.
+
+## Implementation details
+
+The driver as of `v0.0.14` adds an option to use gRPC to communicate with the provider. This is an alpha feature and is introduced with a feature flag `--grpc-supported-providers`. The `--grpc-supported-providers` is a `;` delimited list of all providers that support gRPC for communication. The driver will communicate with the provider using gRPC only if the provider name is in the list of supported providers in `--grpc-supported-providers`. 
+
+> Example usage: `--grpc-supported-providers=provider1;provider2`
+
+To implement a secrets-store-csi-driver provider, you can develop a new provider gRPC server using the stub file available for Go.
+- Use the functions and data structures in the stub file: [service.pb.go](../provider/v1alpha1/service.pb.go) to develop the server code
+  - The stub file and proto file are shared and hosted in the driver. Vendor-in the stub file and proto file in the provider
+  - [fake server example](../provider/fake/fake_server.go)
+- Provider runs as a daemonset and is deployed on the same host(s) as the secrets-store-csi-driver pods
+- Provider unix socket volume path
```

Comment:
Having an example here of the path would help. I believe it should be a configurable base directory in the host filesystem, currently `--provider-volume` on the driver, and the filename should be `provider-name.unix` so for azure:

```
/etc/kubernetes/secrets-store-csi-providers/azure.unix
```

Reply:
That's correct! I've added an example format in the doc now.